src/main/java/org/spdx/sbom/gradle/SpdxSbomTask.java

```diff
@@ -84,6 +97,37 @@ public abstract class SpdxSbomTask extends DefaultTask {
 
   @TaskAction
   public void generateSbom() throws Exception {
+    Set<ComponentIdentifier> componentIds = new HashSet<>();
+    for (var rootComponent : getRootComponents().get()) {
+      componentIds.addAll(
+          gatherSelectedDependencies(rootComponent, new HashSet<>(), new HashSet<>()));
+    }
+
+    Map<ComponentArtifactIdentifier, File> resolvedArtifactsById = new HashMap<>();
+
+    @SuppressWarnings("unchecked")
+    List<ResolvedArtifactResult> resolvedArtifactResults =
```

Comment:
I guess fundamentally, is this okay by gradle's build conventions? Resolving artifacts at task execution time?

Reply:
I've been using this in my project for ~8 months and it's been working great. I think it just looks at the result of the resolution that is done before execution time. The javadoc for `ArtifactResolutionQuery` shows an example of using it in a task.

Reply:
So further research is telling me this is not compatible with task configuration avoidance, someone from game recommended against this approach

Reply:
I'm pretty sure Gradle uses this approach themselves, and I've been using it with task configuration avoidance for a while now. I can look for more info from Gradle on this.

Reply:
gotcha, I'll see if I can validate what I've been told too

Reply:
[From a configuration cache perspective it should be fine](https://gradle-community.slack.com/archives/C013WEPGQF9/p1693327433481099?thread_ts=1693327433.481099&cid=C013WEPGQF9)

Waiting to hear from someone at Gradle conclusively about usage of the API in general, but in this case the existing code is using the same underlying API, and it won't work with projects using variants, so it's probably fine.

Reply:
I haven't looked very closely, but Gradle has been thinking about disallowing resolving dependencies at configuration time: https://github.com/gradle/gradle/issues/2298 . Are there any other times at which we might resolve these artifacts? (Resolving dependencies during task execution makes sense to me)

Reply:
I haven't heard from anyone at gradle directly, but it looks like they're working on a new API too address this. Until that's released this is one of at least two options.

The other option is something that CashApp is doing in [Licensee](https://github.com/cashapp/licensee) where they used detached configurations and copy all the variant information into it. I think the sentiment is that this solution is more "correct" (despite being a little hacky), but it is vastly more complex.